hive-metastore/src/main/java/org/apache/iceberg/hive/CachedClientPool.java

```diff
@@ -53,12 +70,14 @@ public class CachedClientPool implements ClientPool<IMetaStoreClient, TException
             properties,
             CatalogProperties.CLIENT_POOL_CACHE_EVICTION_INTERVAL_MS,
             CatalogProperties.CLIENT_POOL_CACHE_EVICTION_INTERVAL_MS_DEFAULT);
+    this.keySuppliers =
```

Comment:
While I like the keySupplier idea, wouldn't it be much simpler if we just use a static map of suppliers, like:

```
static final Map<String, Supplier> keySuppliers = ImmutableMap.of(
   "uri", () -> {
                try {
                  return UserGroupInformation.getCurrentUser();
                } catch (IOException e) {
                  throw new UncheckedIOException(e);
                }
              }),
   "user_name" -> {
            () -> {
                try {
                  String userName = UserGroupInformation.getCurrentUser().getUserName();
                  return UserNameElement.of(userName);
                } catch (IOException e) {
                  throw new UncheckedIOException(e);
                }
              });
}
```

(as i dont see any need to generate this on the fly.)  And then here in CachedClientPool CTOR, we can just use key to use the conf to make a key?

Reply:
The reason why I use Supplier is to get the current user whenever `CachedClientPool::clientPool` is called. If a `CachedClientPool` is not meant to be shared among different users, I think we don't need to keep the suppliers at all, and just generate the Key instance in CTOR. WDYT?

Reply:
Yea this trips me up every time, but I see, one HiveCatalog making one CachedClientPool.  So I think, we are safe and can make a key in the CTOR, and will be much simpler. @pvary for sanity check here.

Reply:
Thanks, my thought is, lets keep it simple until we have some use case where a process start sharing the HiveCatalog.  I dont see any huge problem just creating one on the fly.